games/applications/run_qpe.py

```diff
@@ -0,0 +1,149 @@
+# ------------------------------------------------------------------------------
+# Part of Red Queen Project.  This file is distributed under the Apache License.
+# See accompanying file /LICENSE for details.
+# ------------------------------------------------------------------------------
+
+# My code was inspired from the paper cited above. It provided a lot of directionality for the
+# structure of the code, however nothing is directly copied from it.
+# https://arxiv.org/pdf/2110.03137.pdf
+
+"""Quantum Phase Estimation"""
+
+import os
+import math
+import pytest
+from qiskit import QuantumCircuit, QuantumRegister, ClassicalRegister
+from applications import backends, run_qiskit_circuit
+
+
+QASM_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), "qasm/qpe/")
+SECRET_ANGLE = 1 / 8
+NUMQUBITS = 3
+
+# qpe3 = Quantum_Phase_Estimation(NUMQUBITS,SECRET_ANGLE)
+# aer_sim = Aer.get_backend('aer_simulator')
+# shots = 4096
+# t_qpe3 = transpile(qpe3, aer_sim)
+# qobj = assemble(t_qpe3, shots=shots)
+# results = aer_sim.run(qobj).result()
+# answer = results.get_counts()
+
+
+# This will convert the fraction into binary for the secret angle
+def fraction_bin(num, precision=10):
+    binary = ""
+    while precision and num != 0:
+        num *= 2
+        bit = int(num)
+        if bit == 1:
+            num -= bit
+            binary += "1"
+        else:
+            binary += "0"
+        precision -= 1
+    return binary
+
+
+# The inverse QFT function was taken from the qiskit textbook from the section discussing
```

Comment:
Emphasize where you got this function at the beginning of your code as well. 

Refer to Caleb's grover's pull request for additional guidance.